server/database/controllers/index.ts

```diff
@@ -0,0 +1,86 @@
+const models = require('../models/index.ts');
+
+type UserData = {
+  userName: string;
+  email: string;
+  hashedPassword: string;
+  followingList: number[];
+  watchedVideos: number[];
+  recommendedVideos: number[];
+  ownedServices: string[];
+};
+
+//!==============================================//
+//!================ USER TABLE ==================//
+//!==============================================//
+
+const addUser = (userData: UserData) => {
+  const newUser = new models.UserTable({
+    userName: userData.userName,
+    email: userData.email,
+    hashedPassword: userData.hashedPassword,
+    followingList:  userData.followingList || [],
+    watchedVideos: userData.watchedVideos || [],
+    recommendedVideos: userData.recommendedVideos || [],
+    ownedServices: userData.ownedServices || [],
+  })
+  return newUser.save()
+    .then(() => {
+      console.log('success posting new user');
+    })
+    .catch((error: any) => {
+      console.log('Error posting new user', error);
+    });
+};
+
+const findUser = (userName: any) => {
+  return models.UserTable.find({'userName': userName})
```

Comment:
I think for authentication purposes, it might be easier to find user by email but for the friends list component finding  by userName will make more sense.

Reply:
I could probably figure out a way to for it to read either, just let me know if you end up needing it to search by email!